server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -251,6 +251,11 @@ public IndexScopedSettings copy(Settings settings, IndexMetadata metadata) {
         return new IndexScopedSettings(settings, this, metadata);
     }
 
+    public boolean registerSetting(Setting<?> setting) {
+        validateSettingKey(setting);
```

Comment:
No need to call `validateSettingsKey`, it is called by `AbstractScopedSettings` already:  https://github.com/opensearch-project/OpenSearch/pull/5347/files#diff-edf64b791aed5932da1f53fa264e42e98bd473f585ceed12892b66cda79c3fd7R154

Reply:
The validateSettingsKey method is overridden in `IndexScopedSettings`.  If it's called in the super wouldn't it only call the validateSettingsKey method in `AbstractScopedSettings`?

Reply:
No, the `validateSettingsKey` will be called for appropriate class: the overridden method in this case.